Framework/DataHandling/src/H5Util.cpp

```diff
@@ -229,6 +229,38 @@ std::string readString(H5::DataSet &dataset) {
   return value;
 }
 
+/**
+ * Returns 1D vector of variable length strings
+ * @param group :: H5::Group already opened
+ * @param name :: name of the dataset in the group (rank must be 1)
+ * @return :: vector of strings
+ */
+std::vector<std::string> readStringVector(Group &group,
+                                          const std::string &name) {
+  hsize_t dims[1];
+  char **rdata;
+  std::vector<std::string> result;
+
+  DataSet dataset = group.openDataSet(name);
+  DataSpace dataspace = dataset.getSpace();
+  DataType datatype = dataset.getDataType();
+
+  int ndims = dataspace.getSimpleExtentDims(dims, NULL);
```

Comment:
Could you do:
`dataspace.getSimpleExtendDims(dims, NULL);` - and ignore the return value? Locally (GCC) this does not emit a warning and it allows us to get rid of the `UNUSED_ARG`